test/api/unit/utils/utils.test.js

```diff
@@ -317,4 +317,73 @@ describe('utils', () => {
       });
     });
   });
+
+  describe('.wait', () => {
+    it('pauses execution for `time` milliseconds', async () => {
+      const time = 100;
+
+      const start = Date.now();
+      await utils.wait(time);
+      const end = Date.now();
+
+      // Some extra padding for execution time
+      expect(end - start).to.within(time, time + 20);
+    });
+  });
+
+  describe('.retry', () => {
+    describe('when `fn` immediately resolves', () => {
+      const value = 'hi';
+      const spy = sinon.spy(() => Promise.resolve(value));
+
+      it('calls `fn` exactly once', async () => {
+        await utils.retry(spy);
+        sinon.assert.calledOnce(spy);
+      });
+
+      it('resolves with the value with which `fn` resolves', async () => {
+        const result = await utils.retry(spy);
+        expect(result).to.eq(value);
+      });
+    });
+
+    describe('when `fn` never resolves', () => {
+      const error = new Error('hi');
+      const spy = sinon.spy(() => Promise.reject(error));
+      const maxAttempts = 3;
+
+      it('calls `fn` `maxAttempts` times', async () => {
+        await utils.retry(spy, { maxAttempts }).catch(err => err);
+        sinon.assert.callCount(spy, maxAttempts);
+      });
+
+      it('rejects with the error with which `fn` rejects', async () => {
+        const result = await utils.retry(spy, { maxAttempts }).catch(err => err);
+        expect(result).to.eq(error);
+      });
+    });
+
+    describe('when `fn` eventually resolves', () => {
+      const value = 'hi';
+      const fn = (() => {
```

Comment:
This is particularly ugly because we are using Sinon 1.x instead of the most recent 9.x where `resolves`, `rejects`, `onFirstCall`, etc are available.